lib/gems/pending/util/postgres_admin.rb

```diff
@@ -136,12 +136,26 @@ def self.backup_pg_compress(opts)
     opts.delete(:dbname)
 
     path = Pathname.new(opts.delete(:local_file))
-
     FileUtils.mkdir_p(path.dirname)
-    Dir.mktmpdir("vmdb_backup", path.dirname) do |dir|
-      runcmd("pg_basebackup", opts, :z => nil, :format => "t", :xlog_method => "fetch", :pgdata => dir)
-      FileUtils.mv(File.join(dir, "base.tar.gz"), path)
-    end
+
+    # Build commandline from AwesomeSpawn
+    args = {:z => nil, :format => "t", :xlog_method => "fetch", :pgdata => "-"}
+    cmd  = AwesomeSpawn.build_command_line("pg_basebackup", combine_command_args(opts, args))
+    $log.info("MIQ(#{name}.#{__method__}) Running command... #{cmd}")
+
+    # Run command in a separate thread
+    read, write   = IO.pipe
+    error_path    = Dir::Tmpname.create("") { |tmpname| tmpname }
+    process       = Process.detach(Kernel.spawn(pg_env(opts), cmd, :out => write, :err => error_path))
```

Comment:
This is interesting. Why the change from `Process.wait` to `Process.detach`?

I thought we wouldn't be able to get the exit status if we detach.

Reply:
`Process.detach` returns a `Process::Waiter` (which is what is also what `Open3.popen3` for example will return instead, for example).

A `Process::Waiter` just inherits from `Thread`, that basically watches a pid until it finishes, and returns a `Process::Status` once it is finished.  The reason I went with this method is I would be wrapping this with `Process::Status` anyway to get the `.exitstatus`, so this is just a easier way of dealing with it.  I am less bothered by the extra thread anyway since I am pretty sure it is well optimized in C anyway, and will mostly deferring to the other IO heavy threads.

Reply:
Okay, that's fine. I just couldn't find the behavior documented anywhere. Can we just use better names for the variables though? Could this one be `wait_thread` or something?